actionpack/test/dispatch/prefix_generation_test.rb

```diff
@@ -327,6 +347,19 @@ def setup
       assert_equal "http://www.example.com/awesome/blog/posts/1", path
     end
 
+    test "[OBJECT] generating engine's route with polymorphic_url from singular resource" do
+      path = engine_object.polymorphic_path(Comment.new)
+      assert_equal "/awesome/blog/comment", path
+
+      path = engine_object.polymorphic_url(Comment.new, host: "www.example.com")
+      assert_equal "http://www.example.com/awesome/blog/comment", path
+    end
+
+    test "[OBJECT] generating engine's route with persists record from singular resource" do
+      comment = Comment.new
+      assert_equal true, comment.persisted?
+    end
```

Comment:
Sorry for the misunderstanding, but this test doesn't test prefix generation, so I think we can just remove it. The test above already handles all relevant cases.

```suggestion
```

Reply:
Woohoo! 🥳 No worries at all, I appreciate your time in looking at this.